### ISSUE TITLE
Burned 1 asset but it burned all 100

### ISSUE THREAD

**Issue by dminer3080** [iron-fish/ironfish#3048]
### What happened?

I minted 100 assets and went to burn 1 but it took all 100 of my assets when I check my wallet:balances. 
![Screenshot 2023-01-20 185231Burn](https://user-images.githubusercontent.com/98119943/213826892-e8259b06-3656-44ae-a144-5f8446359e30.png)

### Version

0.1.62

### Debug output

```shell
dbaker2690@PC:~$ ironfish debug
Database in use, skipping output that requires database.
Stop the node and run the debug command again to show full output.
  status      Show the status of the node
Iron Fish version               0.1.62 @ 4ab43f3
Iron Fish library               0.0.39 @ 4ab43f3estnet
Operating system                Linux x64
CPU model(s)                    12th Gen Intel(R) Core(TM) i9-12900K
CPU threads                     24
RAM total                       7.59 GiB
Heap total                      2.00 GiB
Node version                    v18.13.0
ironfish in PATH                true
Garbage Collector Exposed       false
Telemetry enabled               true
Node name                       Dminer3080
Block graffiti                  Dminer3080
```

### Relevant log output

```shell
dbaker2690@PC:~$ ironfish wallet:mint
Do you want to create a new asset (Y/N)?: y
Enter the name for the new asset: Dminer
Enter metadata for the new asset: golden spendable crypto
Enter the amount to mint in the custom asset: 100
Enter the fee amount in $IRON (min: 0.00000001) [0.00000001]:

You are about to mint Name: Dminer Metadata: golden spendable crypto
100.00000000 plus a transaction fee of $IRON 0.00000001 with the account default

* This action is NOT reversible *

Do you confirm (Y/N)?: y
Creating the transaction: [████████████████████████████████████████] 100% | ETA: 0s

Minted asset Dminer from default
Asset Identifier: 69adeeb76a91c914871679c0f9cadeeb2c2ec2a7dc5d259b4fee5866515aeea9
Value: 100.00000000

Transaction Hash: 55ffba4d15b0280efed53698707a299574ab19fe5fff10b918d542ea5e5c8e91

Find the transaction on https://explorer.ironfish.network/transaction/55ffba4d15b0280efed53698707a299574ab19fe5fff10b918d542ea5e5c8e91 (it can take a few minutes before the transaction appears in the Explorer)
dbaker2690@PC:~$ ironfish wallet:balance
Account: default
Balance: $IRON 0.00000105
dbaker2690@PC:~$ ironfish wallet:balance
Account: default
Balance: $IRON 0.00000104
dbaker2690@PC:~$ ironfish wallet:balances
Account: default
 Asset Name Asset Id                                                         Confirmed Balance
 ────────── ──────────────────────────────────────────────────────────────── ─────────────────
 Dminer     69adeeb76a91c914871679c0f9cadeeb2c2ec2a7dc5d259b4fee5866515aeea9 100.00000000
 $IRON      d7c86706f5817aa718cd1cfad03233bcd64a7789fd9422d3b17af6823a7e6ac6 0.00000104
dbaker2690@PC:~$ ironfish wallet:burn
? Select the asset you wish to burn 69adeeb76a91c914871679c0f9cadeeb2c2ec2a7dc5d259b4fee5866515aeea9 (Dminer)
Enter the amount to burn in the custom asset: 1
Enter the fee amount in $IRON (min: 0.00000001) [0.00000001]:

You are about to burn:
69adeeb76a91c914871679c0f9cadeeb2c2ec2a7dc5d259b4fee5866515aeea9 1.00000000 plus a transaction fee of $IRON 0.00000001 with the account default

* This action is NOT reversible *

Do you confirm (Y/N)?: y
Creating the transaction: [████████████████████████████████████████] 100% | ETA: 0s

Burned asset 69adeeb76a91c914871679c0f9cadeeb2c2ec2a7dc5d259b4fee5866515aeea9 from default
Value: 1.00000000

Transaction Hash: d863a4e6327a4cea5be60bfc5645954b6047a8520bd8ee565c342e2e1cc95a36

Find the transaction on https://explorer.ironfish.network/transaction/d863a4e6327a4cea5be60bfc5645954b6047a8520bd8ee565c342e2e1cc95a36 (it can take a few minutes before the transaction appears in the Explorer)
dbaker2690@PC:~$ ironfish wallet:balances
Account: default
 Asset Name Asset Id                                                         Confirmed Balance
 ────────── ──────────────────────────────────────────────────────────────── ─────────────────
 Dminer     69adeeb76a91c914871679c0f9cadeeb2c2ec2a7dc5d259b4fee5866515aeea9 100.00000000
 $IRON      d7c86706f5817aa718cd1cfad03233bcd64a7789fd9422d3b17af6823a7e6ac6 0.00000104
dbaker2690@PC:~$ ironfish wallet:balances
Account: default
 Asset Name Asset Id                                                         Confirmed Balance
 ────────── ──────────────────────────────────────────────────────────────── ─────────────────
 Dminer     69adeeb76a91c914871679c0f9cadeeb2c2ec2a7dc5d259b4fee5866515aeea9 0.00000000
 $IRON      d7c86706f5817aa718cd1cfad03233bcd64a7789fd9422d3b17af6823a7e6ac6 0.00000099
dbaker2690@PC:~$ ironfish wallet:balances
Account: default
 Asset Name Asset Id                                                         Confirmed Balance
 ────────── ──────────────────────────────────────────────────────────────── ─────────────────
```

### Graffiti

https://testnet.ironfish.network/users/18111

**Comment by hughy** [comment]
There is a bug in the way that the "confirmed" balance is calculated. The balance should resolve after some time, but may temporarily display an incorrect value.

**Comment by hughy** [comment]
Resolved by #3059
